File: Amax-Unity-as-Library_Android/app/src/main/java/com/amax/unityaslibrarydemo/UnityNavigationManager.java
package com.amax.unityaslibrarydemo;

import com.unity3d.player.UnityPlayer;

public class UnityNavigationManager {

    private UnityPlayer unityPlayer;

    private final String GO_NAME = "NavigationManager";
    private final String METHOD_NAME__OPEN_SCENE = "OpenSceneByName";
    private final String METHOD_NAME__SET_NAVIGATION_MODE = "SetNavigationModeByName";
    private final String METHOD_NAME__SET_SHOW_CLOSE_BUTTON = "SetShowCloseButton";

    public UnityNavigationManager(UnityPlayer unityPlayer) {
        this.unityPlayer = unityPlayer;
    }

    public void openScene(EUnityScene scene) {
        unityPlayer.UnitySendMessage(GO_NAME, METHOD_NAME__OPEN_SCENE, scene.toString());
    }

    public void setNavigationMode(EUnityNavigationMode mode) {
        unityPlayer.UnitySendMessage(GO_NAME, METHOD_NAME__SET_NAVIGATION_MODE, mode.toString());
    }

    public void setShowCloseButton(boolean value) {
        unityPlayer.UnitySendMessage(GO_NAME, METHOD_NAME__SET_SHOW_CLOSE_BUTTON, String.valueOf(value));
    }

    public void configure(EUnityNavigationMode navigationMode, EUnityScene scene, boolean showCloseButton) {
        setNavigationMode(navigationMode);
        openScene(scene);
        setShowCloseButton(showCloseButton);
    }

    public enum EUnityNavigationMode
    {
        Default,
        SingleScreen
    }

    public enum EUnityScene
    {
        MainMenu,
        ARObjects,
        ARFaceRecognition
    }

}

File: Amax-Unity-as-Library_iOS/Amax-Unity-as-Library_iOS/UnityNavigationManager.swift
//
//  UnityNavigationManager.swift
//  UIKitPlayground
//
//  Created by Andrei Maksimovich on 20/06/2022.
//

import Foundation
import UnityFramework

class UnityNavigationManager {
    
    private var unityFramework : UnityFramework?
    
    private let GameObjectName: String = "NavigationManager";
    private let MethodNameOpenScene: String = "OpenSceneByName";
    private let MethodNameSetNavigationMode: String = "SetNavigationModeByName";
    private let MethodNameSetShowCloseButton: String = "SetShowCloseButton";
    
    init(ufm: UnityFramework) {
        unityFramework = ufm
    }
    
    func openScene(scene: UnityScene) {
        unityFramework?.sendMessageToGO(withName: GameObjectName, functionName: MethodNameOpenScene, message: scene.rawValue)
    }
    
    func setNavigation(mode: UnitynavigationMode) {
        unityFramework?.sendMessageToGO(withName: GameObjectName, functionName: MethodNameSetNavigationMode, message: mode.rawValue)
    }
    
    func setShowCloseButton(value: Bool) {
        unityFramework?.sendMessageToGO(withName: GameObjectName, functionName: MethodNameSetShowCloseButton, message: value ? "true" : "false")
    }
        
    enum UnityScene: String {
        case MainMenu
        case ARObjects
        case ARFaceRecognition
    }
        
    enum UnitynavigationMode: String {
        case Default
        case SingleScreen
    }
    
}
